scripts/check_duplicates.py

```diff
@@ -0,0 +1,174 @@
+            rds = RosDistroSource(distro)
+            rosdep_data = get_gbprepo_as_rosdep_data(distro)
+            sources.append(CachedDataSource('yaml', distfile, [distro], rosdep_data))
+    for filename in os.listdir(os.path.join(basedir,'rosdep')):
+        if filename.endswith('yaml'):
+            filepath = os.path.join(basedir, 'rosdep', filename)
+            with open(filepath) as f:
+                rosdep_data = yaml.load(f.read())
+            tag = 'osx' if 'osx-' in filepath else ''
+            sources.append(CachedDataSource('yaml', 'flie:/'+filepath, [tag], rosdep_data))
+    return sources
+
+
+def check_duplicates(sources):
+    ret = True
+    # output debug info
+    print_test("checking sources")
+    for source in sources:
+        print_test("- %s" % source.url)
+
+    # create loopkup
+    sources_loader = SourcesListLoader(sources)
+    lookup = RosdepLookup.create_from_rospkg(sources_loader=sources_loader)
+
+    #
+    db_name_view = dict()
+
+    # check if duplicates
+    print_test("checking duplicates")
+    view = lookup.get_rosdep_view(DEFAULT_VIEW_KEY, verbose=None) # to call init
+    for view_key in lookup.rosdep_db.get_view_dependencies(DEFAULT_VIEW_KEY):
+        db_entry=lookup.rosdep_db.get_view_data(view_key)
+        print_test("* %s" % view_key)
+        for dep_name, dep_data in db_entry.rosdep_data.items():
+            if dep_name in db_name_view:
+                print_err("%s is multiply defined in\n\t%s and \n\t%s\n"%(dep_name, db_name_view[dep_name], view_key))
+                ret = False
+            db_name_view[dep_name] = view_key
+    return ret
+
+def main(infile):
+
+    # surces
+    sources = create_default_sources()
+    matcher = DataSourceMatcher.create_default()
+
+    print_test("default sources")
+    for source in sources:
+        print_test("- %s" % source.url)
+
+    # replace with infile
+    for filename in infile:
+        filepath = os.path.join(os.getcwd(), filename)
+        with open(filepath) as f:
+            rosdep_data = yaml.load(f.read())
+        # osx-homebrow uses xos tag
+        tag = 'osx' if 'osx-' in filepath else ''
+        model = CachedDataSource('yaml', 'flie:/'+filepath, [tag], rosdep_data)
+        # add sources if not exists
+        if not [x for x in sources if os.path.basename(filename) == os.path.basename(x.url)]:
+            sources.append(model)
+        else:
+            # remove files with same filename
+            sources = [model if os.path.basename(filename) == os.path.basename(x.url) else x for x in sources]
+
+    ret = True
+    for tag in [['indigo', 'ubuntu', 'trusty'],
+                ['jade', 'ubuntu', 'trusty'],
+                ['kinetic', 'ubuntu', 'xenial'],
+                ['lunar', 'ubuntu', 'xenial']]:
+        matcher.tags = tag
+        print_test('checking with %s'%(matcher.tags))
+        sources = [x for x in sources if matcher.matches(x)]
+        ret = ret & check_duplicates(sources)
+    return ret
```

Comment:
@dirk-thomas yes and I called `main` as `check_duplicates`at the last of `rosdep_duplicates_test.py`
https://github.com/ros/rosdistro/pull/15263/files#diff-857189dd46c3a32ffe6a1424eaf41c5fR5

I just followed https://github.com/ros/rosdistro/blob/master/test/rosdep_formatting_test.py and https://github.com/ros/rosdistro/blob/master/scripts/check_rosdep.py style, so please let me know your recommended test and script code, I can catch your direction.